google-cloud-error_reporting/lib/google/cloud/error_reporting/error_event.rb

```diff
@@ -216,7 +216,7 @@ def self.from_exception exception
           message = exception.message
 
           unless backtrace.nil?
-            error_location = backtrace.first
+            error_location = String backtrace.first
```

Comment:
In cases where `backtrace.first` is `nil` (and presumably `backtrace` is an empty array, as in issue #2149), the value of `message` will be changed from `exception.message` to `": #{exception.message} (#{exception.class})\n\t"`. At least that's how it looks to me. Is that desired?

Reply:
@dazuma Can you weigh in on this?

Reply:
I think @quartzmo brings up a good point, however I don't know for sure what the expected behavior should be in these cases. Of the two suggestions I gave, casting the first backtrace element to a string, and casting a `nil` backtrace to an array, the second seems to preserve more of the original intent in that it doesn't change `message` when a backtrace value is not present.

Reply:
There isn't a requirement for exact behavior, nor do we need to preserve the current syntax. However, I think we can do better than `": #{exception.message} (#{exception.class})\n\t"` which is a bit ugly. My suggestion is to choose one of:
1. blowmage's second option which does not affect the message for an empty backtrace (i.e. the same behavior as nil backtrace)
2. Still treat the two cases (nil and empty backtrace) in the same way, but change the message to something reasonable for that case, like `"#{exception.message} (#{exception.class})"`.

I don't have a strong preference either way. @SpringMT if you like to see the exception class in the resulting string (which seems like a good idea) then option 2 above would do that, _and_ also give the same benefit for the case of nil backtrace.

Reply:
I think an exception class is important to fix error when backtrace dose not exist. 
So, I suggest, when backtrace is nil or backtrace is an empty array, error message is `"#{exception.message} (#{exception.class})"`.

How about this path?

```diff
--- a/google-cloud-error_reporting/lib/google/cloud/error_reporting/error_event.rb
+++ b/google-cloud-error_reporting/lib/google/cloud/error_reporting/error_event.rb
@@ -213,12 +213,12 @@ module Google
         #   from the given exception.
         def self.from_exception exception
           backtrace = exception.backtrace
-          message = exception.message
+          message = "#{exception.message} (#{exception.class})"
 
-          unless backtrace.nil?
-            error_location = String backtrace.first
+          if !backtrace.nil? && !backtrace.empty?
+            error_location = backtrace.first
 
-            message = "#{error_location}: #{message} (#{exception.class})\n\t" +
+            message = "#{error_location}: #{message}\n\t" +
                       backtrace.drop(1).join("\n\t")
             file_path, line_number, function_name = error_location.split(":")
             function_name = function_name.to_s[/`(.*)'/, 1]
```

Reply:
@dazuma ^ thoughts?

Reply:
Looks good to me.

Reply:
@dazuma I agree. @SpringMT Can you make your proposed patch change in this PR?

Reply:
Done! 
https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2152/commits/0e79f886e0eb40832309de19297191f5c3a45cfd